data-collection/data-collection/View Controllers/Floating Panel View Controller/FloatingPanelViewController.swift

```diff
@@ -0,0 +1,484 @@
+            }
+            else {
+                state = .partial
+            }
+        }
+        else if gestureRecognizer.state != .cancelled {
+            // Calculate the new constraint constant, based on changes from the initial constant.
+            var newConstant = initialResizableLayoutConstraintConstant + (isCompactWidth ? -translation.y : translation.y)
+            
+            // Limit the constant to be within min/max.
+            newConstant = max(minimumHeight, newConstant)
+            newConstant = min(maximumHeight, newConstant)
+            
+            // Set the new constant on the resizable constraint.
+            resizeableLayoutConstraint.constant = newConstant
+        }
+        else {
+            // On cancellation, animate the panel height to its original size.
+            animateResizableConstrant(initialResizableLayoutConstraintConstant)
+        }
+    }
+    
+    /// Updates the handlebar visibility based on horizontal size class
+    /// and the value of `allowManualResize`.
+    private func updateHandlebarVisibility() {
+        topHandlebarView.isHidden = !allowManualResize || !isCompactWidth
+        bottomHandlebarView.isHidden = !allowManualResize || isCompactWidth
+    }
+    
+    /// Handles resizing the panel when the user resizes via a "flick".
+    /// - Parameter velocity: The velocity of the gesture.
+    private func handleFlick(_ velocity: CGPoint) {
+        if isCompactWidth {
+            if velocity.y > 0 {
+                // Velocity > 0 means the user is dragging the view shorter.
+                // Switch to either .minimized or .partial depending on the constraint constant value.
+                state = resizeableLayoutConstraint.constant <= internalPartialHeight ? .minimized : .partial
+            }
+            else {
+                // The user is dragging the view larger.
+                // Switch to either .partial or .full depending on the constraint constant value.
+                state = resizeableLayoutConstraint.constant <= internalPartialHeight ? .partial : .full
+            }
+        }
+        else {
+            // Regular width size class.
+            if velocity.y > 0 {
+                // Velocity > 0 means the user is dragging the view larger.
+                // Switch to either .partial or .full depending on the constraint constant value.
+                state = resizeableLayoutConstraint.constant <= internalPartialHeight ? .partial : .full
+            }
+            else {
+                // The user is dragging the view shorter.
+                // Switch to either .minimized or .partial depending on the constraint constant value.
+                state = resizeableLayoutConstraint.constant <= internalPartialHeight ? .minimized : .partial
+            }
+        }
+    }
+    
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
```

Comment:
![image](https://user-images.githubusercontent.com/2257493/89356141-e7fd3100-d671-11ea-92d1-946bc454cb7c.png)
```suggestion
    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
```